node/services/service.go

```diff
@@ -31,14 +29,32 @@ func HeaderSyncer(cfg Config) func(ex header.Exchange, store header.Store) (*hea
 	}
 }
 
+// P2PSubscriber creates a new header.P2PSubscriber.
+func P2PSubscriber(sub *pubsub.PubSub, syncer *header.Syncer) *header.P2PSubscriber {
+	return header.NewP2PSubscriber(sub, syncer.Validate)
+}
+
 // HeaderService creates a new header.Service.
-func HeaderService(lc fx.Lifecycle, syncer *header.Syncer, sub *pubsub.PubSub) (*header.Service, header.Broadcaster) {
-	service := header.NewHeaderService(syncer, sub)
+func HeaderService(lc fx.Lifecycle,
+	syncer *header.Syncer,
+	psManager *header.P2PSubscriber,
+	ex header.Exchange,
+) *header.Service {
+
+	headerLifecycles := []header.Lifecycle{
+		syncer,
+		psManager,
+		ex,
+	}
+	// TODO @renaynay: implement adding listener to headerLifecycles if core is enabled
```

Comment:
FYI this is waiting on broadcaster to be based on the refactoring so it can be implemented.